### DISCUSSION TITLE
Minimal repro replacing nvim-cmp is not triggering auto-completion

### DISCUSSION THREAD

**Discussion by idelice** [LazyVim/LazyVim#4623] [General]
I want to test out blink.nvim in a minimal repo, so I added this to my `lazy.lua` file:

```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
  spec = {
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    {
      "saghen/blink.cmp",
      lazy = false, -- lazy loading handled internally
      dependencies = "rafamadriz/friendly-snippets",
      version = "v0.*",
    },
    {
      "hrsh7th/nvim-cmp",
      enabled = false,
    },
  },
})
```

After opening a file (tsx/java), i get no completion.

If i remove the `nvim-cmp` part or set `enabled = true`, then `nvim-cmp` is basically taking over the auto-completion.

How do I disable `nvim-cmp` in favor of `blink.lua` for this purpose?

**Comment by dotfrag** [comment]
I'm not on pc to test, but it looks like you are missing opts or config to trigger the setup function.

Correct, adding `opts = {}` works.

**Reply by idelice** [reply]
> opts = {}

Please give me the full example on how you made it work.. opts = {} didn't do the trick but i may have done it wrongly

**Reply by dotfrag** [reply]
```lua
{
  "saghen/blink.cmp",
  lazy = false, -- lazy loading handled internally
  dependencies = "rafamadriz/friendly-snippets",
  version = "v0.*",
  opts = {},
},
{
  "hrsh7th/nvim-cmp",
  enabled = false,
},
```

This is needed because when you add a plugin, you need to run it's setup function (almost always called `setup`, but some plugins might have something else).

The reason the above works, is because lazy (the package manger), will automatically call the `setup()` function of the plugin when `opts` exists, or if `config` is set to true. More info [here](https://lazy.folke.io/spec), specifically:

> config is executed when the plugin loads. The default implementation will automatically run require(MAIN).setup(opts) if opts or config = true is set.

**Reply by idelice** [reply]
Thank you! It did the trick
